paasta_tools/setup_kubernetes_job.py

```diff
@@ -233,6 +233,8 @@ def setup_kube_deployments(
     eks: bool = False,
 ) -> bool:
 
+    existing_apps = set()
+    existing_kube_deployments = set()
     if service_instance_configs_list:
```

Comment:
Maybe we should flip this to:

```
if not service_instance_configs_list:
    return True

existing_kube_deployments = set(list_all_paasta_deployments(kube_client))
existing_apps = {
    (deployment.service, deployment.instance, deployment.namespace)
    for deployment in existing_kube_deployments
}
```

that way you don't need the empty `set()`s

Reply:
I don't think we should call list_all_paasta_deployments() which calls the k8s API and gets all deployments on every namespace if `service_instance_configs_list` is None or empty. 

Type hinting the empty sets sound like a better solution to me, since the only reason why I moved the definition out is cuz it was created inside an if block but referenced later outside the if

Reply:
maybe something like:

```
 existing_apps: Set[Tuple[str, str, str]] = set()
 existing_kube_deployments: Set[List[KubeDeployment]] = set()
```